From abe5c5f1ad5575d04e53a91d5c2ead2bb50292a7 Mon Sep 17 00:00:00 2001
From: Randy Dunlap <rdunlap@infradead.org>
Date: Sun, 19 Jul 2026 15:32:31 -0700
Subject: [PATCH 08438/15333] bus: mhi: Clean up some kernel-doc warnings

Clean up some kernel-doc comments and warnings:
- use correct format for struct members
- add one struct member description
- add one function parameter description
- mark one enum as private
- add a leading '*' on one kernel-doc line

Fixes these warnings:
Warning: include/linux/mhi.h:108 struct member 'target_link_speed' not
 described in 'mhi_link_info'
Warning: include/linux/mhi.h:108 struct member 'target_link_width' not
 described in 'mhi_link_info'
Warning: ../include/linux/mhi.h:159 Enum value 'MHI_STATE_MAX' not
 described in enum 'mhi_state'
Warning: ../include/linux/mhi.h:212 bad line: for UL channels, multiple
 of 8 ring elements for DL channels
Warning: ../include/linux/mhi.h:236 struct member 'wake_capable' not
 described in 'mhi_channel_config'
Warning: ../include/linux/mhi.h:449 struct member 'M0' not described
 in 'mhi_controller'
Warning: ../include/linux/mhi.h:449 struct member 'M2' not described
 in 'mhi_controller'
Warning: ../include/linux/mhi.h:449 struct member 'M3' not described
 in 'mhi_controller'
Warning: ../include/linux/mhi.h:528 struct member 'id_table' not described
 in 'mhi_driver'
Warning: ../include/linux/mhi.h:543 function parameter 'mhi_cntrl' not
 described in 'mhi_free_controller'

Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@oss.qualcomm.com>
---
 include/linux/mhi.h | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/include/linux/mhi.h b/include/linux/mhi.h
index 0d60058bf5ae..ca24ca6c6a48 100644
--- a/include/linux/mhi.h
+++ b/include/linux/mhi.h
@@ -116,8 +116,8 @@ struct image_info {
 
 /**
  * struct mhi_link_info - BW requirement
- * target_link_speed - Link speed as defined by TLS bits in LinkControl reg
- * target_link_width - Link width as defined by NLW bits in LinkStatus reg
+ * @target_link_speed: Link speed as defined by TLS bits in LinkControl reg
+ * @target_link_width: Link width as defined by NLW bits in LinkStatus reg
  */
 struct mhi_link_info {
 	unsigned int target_link_speed;
@@ -172,6 +172,7 @@ enum mhi_state {
 	MHI_STATE_M3_FAST = 0x6,
 	MHI_STATE_BHI = 0x7,
 	MHI_STATE_SYS_ERR = 0xFF,
+	/* private: */
 	MHI_STATE_MAX,
 };
 
@@ -226,12 +227,12 @@ enum mhi_db_brst_mode {
  * @type: Channel type
  * @ee_mask: Execution Environment mask for this channel
  * @pollcfg: Polling configuration for burst mode.  0 is default.  milliseconds
-	     for UL channels, multiple of 8 ring elements for DL channels
+ *	     for UL channels, multiple of 8 ring elements for DL channels
  * @doorbell: Doorbell mode
  * @lpm_notify: The channel master requires low power mode notifications
  * @offload_channel: The client manages the channel completely
  * @doorbell_mode_switch: Channel switches to doorbell mode on M0 transition
- * @wake-capable: Channel capable of waking up the system
+ * @wake_capable: Channel capable of waking up the system
  */
 struct mhi_channel_config {
 	char *name;
@@ -349,7 +350,9 @@ struct mhi_controller_config {
  * @dev_state: MHI device state
  * @dev_wake: Device wakeup count
  * @pending_pkts: Pending packets for the controller
- * @M0, M2, M3: Counters to track number of device MHI state changes
+ * @M0: Counter to track number of device MHI state changes
+ * @M2: Counter to track number of device MHI state changes
+ * @M3: Counter to track number of device MHI state changes
  * @transition_list: List of MHI state transitions
  * @transition_lock: Lock for protecting MHI state transition list
  * @wlock: Lock for protecting device wakeup
@@ -508,6 +511,7 @@ struct mhi_result {
 
 /**
  * struct mhi_driver - Structure representing a MHI client driver
+ * @id_table: table of MHI channel names that a driver supports
  * @probe: CB function for client driver probe function
  * @remove: CB function for client driver remove function
  * @ul_xfer_cb: CB function for UL data transfer
@@ -539,6 +543,7 @@ struct mhi_controller *mhi_alloc_controller(void);
 
 /**
  * mhi_free_controller - Free the MHI Controller structure
+ * @mhi_cntrl: MHI controller to free
  * Free the mhi_controller structure which was previously allocated
  */
 void mhi_free_controller(struct mhi_controller *mhi_cntrl);
-- 
2.55.0

